include/fastdds/dds/core/status/StatusMask.hpp

```diff
@@ -0,0 +1,291 @@
+    }
+
+    /**
+     * Get the StatusMask associated with dds::core::status::OfferedIncompatibleQosStatus
+     *
+     * @return StatusMask offered_incompatible_qos
+     */
+    inline static StatusMask offered_incompatible_qos()
+    {
+        return StatusMask(0x00000001 << 5u);
+    }
+
+    /**
+     * Get the StatusMask associated with dds::core::status::RequestedIncompatibleQosStatus
+     *
+     * @return StatusMask requested_incompatible_qos
+     */
+    inline static StatusMask requested_incompatible_qos()
+    {
+        return StatusMask(0x00000001 << 6u);
+    }
+
+    /**
+     * Get the StatusMask associated with dds::core::status::SampleLostStatus
+     *
+     * @return StatusMask sample_lost
+     */
+    inline static StatusMask sample_lost()
+    {
+        return StatusMask(0x00000001 << 7u);
+    }
+
+    /**
+     * Get the StatusMask associated with dds::core::status::SampleRejectedStatus
+     *
+     * @return StatusMask sample_rejected
+     */
+    inline static StatusMask sample_rejected()
+    {
+        return StatusMask(0x00000001 << 8u);
+    }
+
+    /**
+     * Get the StatusMask associated with dds::core::status::data_on_readers
+     *
+     * @return StatusMask data_on_readers
+     */
+    inline static StatusMask data_on_readers()
+    {
+        return StatusMask(0x00000001 << 9u);
+    }
+
+    /**
+     * get the statusmask associated with dds::core::status::data_available
+     *
+     * @return statusmask data_available
+     */
+    inline static StatusMask data_available()
+    {
+        return StatusMask(0x00000001 << 10u);
+    }
+
+    /**
+     * Get the StatusMask associated with dds::core::status::LivelinessLostStatus
+     *
+     * @return StatusMask liveliness_lost
+     */
+    inline static StatusMask liveliness_lost()
+    {
+        return StatusMask(0x00000001 << 11u);
+    }
+
+    /**
+     * Get the StatusMask associated with dds::core::status::LivelinessChangedStatus
+     *
+     * @return StatusMask liveliness_changed
+     */
+    inline static StatusMask liveliness_changed()
+    {
+        return StatusMask(0x00000001 << 12u);
+    }
+
+    /**
+     * Get the statusmask associated with dds::core::status::PublicationMatchedStatus
+     *
+     * @return StatusMask publication_matched
+     */
+    inline static StatusMask publication_matched()
+    {
+        return StatusMask(0x00000001 << 13u);
+    }
+
+    /**
+     * Get the statusmask associated with dds::core::status::SubscriptionMatchedStatus
+     *
+     * @return StatusMask subscription_matched
+     */
+    inline static StatusMask subscription_matched()
+    {
+        return StatusMask(0x00000001 << 14u);
+    }
+
+    bool is_active(
+            StatusMask status)
+    {
+        bitset<16> r = *this & status;
+        return r == status;
+    }
+
+    bool is_compatible(
+            StatusMask base) const
+    {
+        return (*this == all() || *this == base);
+    }
```

Comment:
This method can be deleted by the moment, it is there as I simply move this file from PSM API. it will be useful later when StatusCondition are implemented, as you always need to check if the StatusMask associated with an entity has a specific status active or not